Marlin/src/lcd/tft/ui_common.cpp

```diff
@@ -416,6 +456,78 @@ void MarlinUI::clear_lcd() {
   cursor.set(0, 0);
 }
 
+uint8_t _get_word(const char * const string, read_byte_cb_t cb_read_byte, lchar_t &last_char) {
+  if (!string) return 0;
+
+  const uint8_t *p = (uint8_t*)string;
+  lchar_t wc;
+  uint8_t c = 0;
+
+  // find the end of the part
+  for (;;) {
+    p = get_utf8_value_cb(p, cb_read_byte, wc);
+    const bool eol = !wc;         // zero ends the string
+    // End or a break between phrases?
+    if (eol || wc == ' ' || wc == '-' || wc == '+' || wc == '.' || wc == '\n') {
+      c += !eol;                  // +1 so the space will be printed
+      last_char = wc;
+      return c;
+    }
+    else c++;                     // count word characters
+  }
+}
+
+template<typename T>
+void _wrap_string(uint8_t &row, T string, read_byte_cb_t cb_read_byte, const bool flush=false) {
+
+  auto print_str = [&row] () {
+    menu_line(row++);
+    tft_string.trim();
+    tft.add_text(tft_string.center(TFT_WIDTH), MENU_TEXT_Y, COLOR_MENU_TEXT, tft_string);
+    tft_string.set();
+  };
+
+  const uint8_t *p;
+  uint8_t wrd_len = 0;
+  p = (uint8_t*)string;
+
+  lchar_t last_char;
+  p = &p[wrd_len];
+  bool eol = !p;
+  while (!eol) {
+    wrd_len = _get_word((const char *)p, cb_read_byte, last_char);
+    const uint8_t len = tft_string.length;
+    tft_string.add((T)p, wrd_len);
+    const uint32_t wid = tft_string.width();
+
+    if (wid > TFT_WIDTH) {
+      tft_string.truncate(len);
+      print_str();
+      tft_string.set((T)p, wrd_len);
+    }
+    if (last_char == '\n') {
+      tft_string.truncate(tft_string.length - 1);
+      print_str();
+    }
+
+    p = &p[wrd_len];
```

Comment:
This crashes with UTF-8 characters. `_get_word` returns the number of UTF-8 characters, while `p = &p[wrd_len]` advances by the number of bytes.
I guess `p = &p[wrd_len];` should be replaced with something like "advance by wrd_len utf-8 chars, or just return the next pointer from `_get_word`

Reply:
Yes, something like that will be neccessary.